pytensor/tensor/rewriting/linalg.py

```diff
@@ -17,16 +19,40 @@
 logger = logging.getLogger(__name__)
 
 
+def is_matrix_transpose(x: TensorVariable) -> bool:
+    """Check if a variable corresponds to a transpose of the last two axes"""
+    node = x.owner
+    if (
+        node
+        and isinstance(node.op, DimShuffle)
+        and not (node.op.drop or node.op.augment)
+    ):
+        [inp] = node.inputs
+        ndims = inp.type.ndim
+        if ndims < 2:
+            return False
+        transpose_order = tuple(range(ndims - 2)) + (ndims - 1, ndims - 2)
+        return cast(bool, node.op.new_order == transpose_order)
+    return False
+
+
+def T(x: TensorVariable) -> TensorVariable:
```

Comment:
numpy transpose does a different thing

<img width="362" alt="image" src="https://github.com/pymc-devs/pytensor/assets/11705326/0f82ce68-539e-4fb8-b04f-76b4d629d08d">

Reply:
This is just a helper function for the rewrites in this section that only transpose the last two dimensions. Makes the rewrites easier to read

Reply:
I think it should not be there, easy to confuse with numpy transpose and it does it differntly

Reply:
The point was to make the rewrites changes here to track/ review. What before was var.T now is T(var) because we keep working only on the last 2 dimensions. All rewrites work for batches of matrices the linear algebra equivalence still comes from the matrix case.

Happy to add a leading underscore, or call it something more verbose like matrix_T